analyzer.go

```diff
@@ -19,19 +20,87 @@ const (
 )
 
 //nolint:gochecknoglobals // struct is not big, can be skip.
-var Analyzer = &analysis.Analyzer{
-	Name:     "canonicalheader",
-	Doc:      "canonicalheader checks whether net/http.Header uses canonical header",
-	Run:      run,
-	Requires: []*analysis.Analyzer{inspect.Analyzer},
+var Analyzer = New()
+
+func New() *analysis.Analyzer {
+	c := &canonicalHeader{}
+
+	a := &analysis.Analyzer{
+		Name:     "canonicalheader",
+		Doc:      "canonicalheader checks whether net/http.Header uses canonical header",
+		Requires: []*analysis.Analyzer{inspect.Analyzer},
+		Run:      c.run,
+	}
+
+	a.Flags.Init("canonicalheader", flag.ExitOnError)
+	a.Flags.Var(&c.exclusions, "exclusions", "comma-separated list of exclusion rules")
+	a.Flags.BoolVar(&c.useDefaultExclusion, "useDefaultExclusion", true, "use default exclusion rules")
+
+	return a
+}
+
+type canonicalHeader struct {
+	useDefaultExclusion bool
+	exclusions          stringSet
+}
+
+type reasonReport uint8
+
+const (
+	reasonReportNonCanonical reasonReport = iota + 1
+	reasonReportExclusion
+	reasonReportInitialism
+)
+
+func (r reasonReport) String() string {
+	switch r {
+	case reasonReportNonCanonical:
+		return "non-canonical"
+
+	case reasonReportExclusion:
+		return "exclusion"
```

Comment:
This text is displayed I'm the error message, no?

The look at https://github.com/lasiar/canonicalheader/pull/60#discussion_r1835526951

What about this?

```suggestion
		return "excluded"
```

Reply:
I'll simplify the report text.
The reason type will be deleted.